src/test/java/com/commercetools/sync/cartdiscounts/utils/CartDiscountUpdateActionUtilsTest.java

```diff
@@ -0,0 +1,563 @@
+        when(newCartDiscountDraft.getCartPredicate()).thenReturn("1 = 1");
+
+        final Optional<UpdateAction<CartDiscount>> changeCartPredicateUpdateAction =
+            buildChangeCartPredicateUpdateAction(CART_DISCOUNT_WITH_RELATIVE_VALUE, newCartDiscountDraft);
+
+        assertThat(changeCartPredicateUpdateAction).contains(ChangeCartPredicate.of("1 = 1"));
+    }
+
+    @Test
+    public void buildChangeCartPredicateUpdateAction_WithNullValues_ShouldBuildUpdateAction() {
+        final CartDiscountDraft newCartDiscountDraft = mock(CartDiscountDraft.class);
+        when(newCartDiscountDraft.getCartPredicate()).thenReturn(null);
+
+        final Optional<UpdateAction<CartDiscount>> changeCartPredicateUpdateAction =
+            buildChangeCartPredicateUpdateAction(CART_DISCOUNT_WITH_RELATIVE_VALUE, newCartDiscountDraft);
+
+        assertThat(changeCartPredicateUpdateAction).contains(ChangeCartPredicate.of((String) null));
+    }
+
+    @Test
+    public void buildChangeCartPredicateUpdateAction_WithSameValues_ShouldNotBuildUpdateAction() {
+        final CartDiscountDraft newCartDiscountDraft = mock(CartDiscountDraft.class);
+        when(newCartDiscountDraft.getCartPredicate())
+            .thenReturn("totalPrice = \"10.00 EUR\" and (shippingInfo.shippingMethodName = \"FEDEX\")");
+
+        final Optional<UpdateAction<CartDiscount>> changePredicateUpdateAction =
+            buildChangeCartPredicateUpdateAction(CART_DISCOUNT_WITH_RELATIVE_VALUE, newCartDiscountDraft);
+
+        assertThat(changePredicateUpdateAction).isNotPresent();
+    }
+
+    @Test
+    public void buildChangeTargetUpdateAction_WithDifferentLineItemTargetValues_ShouldBuildUpdateAction() {
+        final CartDiscountDraft newCartDiscountDraft = mock(CartDiscountDraft.class);
+        final CartDiscountTarget cartDiscountTarget = LineItemsTarget.of("quantity > 0");
+        when(newCartDiscountDraft.getTarget()).thenReturn(cartDiscountTarget);
+
+        final Optional<UpdateAction<CartDiscount>> changeTargetUpdateAction =
+            buildChangeTargetUpdateAction(CART_DISCOUNT_WITH_RELATIVE_VALUE, newCartDiscountDraft);
+
+        assertThat(changeTargetUpdateAction).contains(ChangeTarget.of(cartDiscountTarget));
+    }
+
+    @Test
+    public void buildChangeTargetUpdateAction_WithSameLineItemTargetValues_ShouldNotBuildUpdateAction() {
+        final CartDiscountDraft newCartDiscountDraft = mock(CartDiscountDraft.class);
+        final CartDiscountTarget cartDiscountTarget = LineItemsTarget.of("1 = 1");
+        when(newCartDiscountDraft.getTarget()).thenReturn(cartDiscountTarget);
+
+        final Optional<UpdateAction<CartDiscount>> changeTargetUpdateAction =
+            buildChangeTargetUpdateAction(CART_DISCOUNT_WITH_RELATIVE_VALUE, newCartDiscountDraft);
+
+        assertThat(changeTargetUpdateAction).isNotPresent();
+    }
+
+    @Test
+    public void buildChangeTargetUpdateAction_WithNullValues_ShouldBuildUpdateAction() {
```

Comment:
```suggestion
    public void buildChangeTargetUpdateAction_WithNullNewValue_ShouldBuildUpdateAction() {
```
What about tests for:

both null values (old and new)?
just old null value?